libbeat/common/file/rotator.go

```diff
@@ -47,20 +51,27 @@ func (rr rotateReason) String() string {
 		return "file size"
 	case rotateReasonManualTrigger:
 		return "manual trigger"
+	case rotateReasonDay:
+		return "day"
 	default:
 		return "unknown"
 	}
 }
 
 // Rotator is a io.WriteCloser that automatically rotates the file it is
-// writing to when it reaches a maximum size. It also purges the oldest rotated
-// files when the maximum number of backups is reached.
+// writing to when it reaches a maximum size and optionally on a daily basis.
+// It also purges the oldest rotated files when the maximum number of backups
+// is reached.
 type Rotator struct {
 	filename     string
 	maxSizeBytes uint
 	maxBackups   uint
 	permissions  os.FileMode
 	log          Logger // Optional Logger (may be nil).
+	daily        bool
+	lastYear     int
```

Comment:
Why do you store date in separately instead of a `time.Time` struct? It would make easier to check if the log needs to be rotated or not by using `Since`.